.changeset/pink-cars-guess.md

```diff
@@ -0,0 +1,6 @@
+---
+"@pnpm/plugin-commands-installation": patch
+"pnpm": patch
+---
+
+Fix a bug in which `pnpm add` downloads packages whose `libc` differ from `pnpm.supportedArchitectures.libc`.
```

Comment:
I didn't want to make [this change](https://github.com/pnpm/pnpm/pull/9745/commits/d5ddf1f688da3f02a8b4276a9238ec0610a9c451) in a separate PR because it would cause conflicts.

`fetchFullMetadata` from `master` did not include `opts.supportedArchitectures?.libc` in its calculation.